api/src/pcapi/routes/backoffice_v3/collective_offer_templates.py

```diff
@@ -223,39 +191,165 @@ def get_reject_collective_offer_template_form(collective_offer_template_id: int)
 @list_collective_offer_templates_blueprint.route("/<int:collective_offer_template_id>/reject", methods=["POST"])
 @utils.permission_required(perm_models.Permissions.FRAUD_ACTIONS)
 def reject_collective_offer_template(collective_offer_template_id: int) -> utils.BackofficeResponse:
-    collective_offer_template = educational_models.CollectiveOfferTemplate.query.get_or_404(
-        collective_offer_template_id
+    success = _batch_reject_collective_offer_templates([collective_offer_template_id])
+    return redirect(
+        request.referrer or url_for("backoffice_v3_web.collective_offer_template.list_collective_offer_templates"),
+        303 if success else 400,
     )
 
-    new_validation_status = OfferValidationStatus.REJECTED
-    if collective_offer_template.validation != OfferValidationStatus.PENDING:
-        flash("Seules les offres collectives vitrines en attente peuvent être rejetées", "warning")
-        return redirect(
-            request.referrer or url_for("backoffice_v3_web.collective_offer_template.list_collective_offer_templates"),
-            303,
+
+def _batch_validate_collective_offer_templates(collective_offer_template_ids: list[int]) -> bool:
+    collective_offer_templates = educational_models.CollectiveOfferTemplate.query.filter(
+        educational_models.CollectiveOfferTemplate.id.in_(collective_offer_template_ids),
+        educational_models.CollectiveOfferTemplate.validation == OfferValidationStatus.PENDING,
+    ).all()
+
+    if len(collective_offer_template_ids) != len(collective_offer_templates):
+        flash("Seules les offres collectives vitrines en attente peuvent être validées", "danger")
+        return False
+
+    collective_offer_template_update_succeed_ids: list[int] = []
+    collective_offer_template_update_failed_ids: list[int] = []
+
+    for collective_offer_template in collective_offer_templates:
+        new_validation_status = OfferValidationStatus.APPROVED
+        collective_offer_template.validation = new_validation_status
+        collective_offer_template.isActive = True
+
+        try:
+            db.session.commit()
+        except Exception:  # pylint: disable=broad-except
+            collective_offer_template_update_failed_ids.append(collective_offer_template.id)
+            break
```

Comment:
```suggestion
            continue
```
Comme tu comptes les erreurs, j'imagine que tu ne souhaites pas arrêter la boucle, mais passer à l'offre suivante